api/project.py

```diff
@@ -41,3 +64,62 @@ def get(self, request) -> JSONResponse:
             "knowledge_base_ids": [str(list.id) for list in project.knowledge_bases],
         }
         return JSONResponse(result)
+
+
+class ProjectCreationFromWorkflow(HTTPEndpoint):
+    async def post(self, request) -> JSONResponse:
+        user_id = request.query_params["user_id"]
+        request_body = await request.json()
+        name = request_body["name"]
+        description = request_body["description"]
+        tokenizer = request_body["tokenizer"]
+        store_id = request_body["store_id"]
+
+        user = auth_manager.get_user_by_id(user_id)
+        organization = auth_manager.get_organization_by_user_id(user.id)
+
+        if not bucket_exists(str(organization.id)):
+            create_bucket(str(organization.id))
+
+        project = project_manager.create_project(
+            organization.id, name, description, user.id
+        )
+        project_manager.update_project(project_id=project.id, tokenizer=tokenizer)
+
+        Session = sessionmaker(get_workflow_db_engine())
+        with Session() as session:
+            results = session.execute(
+                f"SELECT record FROM store_entry WHERE store_id = '{store_id}'"
+            ).all()
+
+        data = [result for result, in results]
+
+        upload_task = upload_task_manager.create_upload_task(
+            user_id=user.id,
+            project_id=project.id,
+            file_name=name,
+            file_type="json",
```

Comment:
use of enum is recommended for constant strings

Reply:
added these enums:
```python
class RecordImportFileTypes(Enum):
    JSON = "JSON"
    CSV = "CSV"
    XLSX = "XLSX"

```